src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -1733,6 +1733,31 @@ Object internalProperties() {
      */
     abstract BoundMethodHandle rebind();
 
+    /* non-public */
+    void maybeCustomize() {
+        if (form.customized == null) {
+            byte count = customizationCount;
+            if (count >= CUSTOMIZE_THRESHOLD) {
+                customize();
+            } else {
+                customizationCount = (byte) (count + 1);
+            }
+        }
+    }
+
+    /** Craft a LambdaForm customized for this particular MethodHandle. */
+    /*non-public*/
+    void customize() {
+        final MethodHandle mh = this;
+        updateForm(new Function<>() {
+                public LambdaForm apply(LambdaForm oldForm) {
+                    return oldForm.customize(mh);
```

Comment:
I think you can drop line 1750 and write this as `return oldForm.customize(MethodHandle.this);`. Simpler and might avoid an additional, implicit capture.